checks.d/http_check.py

```diff
@@ -161,10 +164,15 @@ def __init__(self, name, init_config, agentConfig, instances):
                     user=proxy_settings['user'],
                     password=proxy_settings['password'],
                     uri=uri)
-            self.proxies = {
-                'http': "http://{uri}".format(uri=uri),
-                'https': "https://{uri}".format(uri=uri)
-            }
+            self.proxies['http'] = "http://{uri}".format(uri=uri)
+            self.proxies['https'] = "https://{uri}".format(uri=uri)
+        else:
+            self.proxies['http'] = environ.get('http', None)
+            self.proxies['https'] = environ.get('https', None)
```

Comment:
Why `http` and `https` as the environment variables? Those seem very prone to unintended overlap. 

(The requests library uses [HTTP_PROXY and HTTPS_PROXY](http://docs.python-requests.org/en/v2.6.2/user/advanced/#proxies).)